seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/client/HttpHelper.java

```diff
@@ -0,0 +1,163 @@
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.doris.client;
+
+import org.apache.seatunnel.common.utils.JsonUtils;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.entity.ByteArrayEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.DefaultRedirectStrategy;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+@Slf4j
+public class HttpHelper {
+    private static final int DEFAULT_CONNECT_TIMEOUT = 1000000;
+
+    public HttpEntity getHttpEntity(CloseableHttpResponse resp) {
+        int code = resp.getStatusLine().getStatusCode();
+        if (HttpStatus.SC_OK != code) {
+            log.warn("Request failed with code:{}", code);
+            return null;
+        }
+        HttpEntity respEntity = resp.getEntity();
+        if (null == respEntity) {
+            log.warn("Request failed with empty response.");
+            return null;
+        }
+        return respEntity;
+    }
+
+    public String doHttpGet(String getUrl) throws IOException {
+        log.info("Executing GET from {}.", getUrl);
+        try (CloseableHttpClient httpclient = buildHttpClient()) {
+            HttpGet httpGet = new HttpGet(getUrl);
+            try (CloseableHttpResponse resp = httpclient.execute(httpGet)) {
+                HttpEntity respEntity = resp.getEntity();
+                if (null == respEntity) {
+                    log.warn("Request failed with empty response.");
+                    return null;
+                }
+                return EntityUtils.toString(respEntity);
+            }
+        }
+    }
+
+    public Map<String, Object> doHttpGet(String getUrl, Map<String, String> header) throws IOException {
+        log.info("Executing GET from {}.", getUrl);
+        try (CloseableHttpClient httpclient = HttpClients.createDefault()) {
+            HttpGet httpGet = new HttpGet(getUrl);
+            if (null != header) {
+                for (Map.Entry<String, String> entry : header.entrySet()) {
+                    httpGet.setHeader(entry.getKey(), String.valueOf(entry.getValue()));
+                }
+            }
+            try (CloseableHttpResponse resp = httpclient.execute(httpGet)) {
+                HttpEntity respEntity = getHttpEntity(resp);
+                if (null == respEntity) {
+                    log.warn("Request failed with empty response.");
+                    return null;
```

Comment:
Throw exception here? I think if reqeust failed the business whole process should be stopped.

Reply:
User can set maxRetry for multiple attempts.

Reply:
Where?

Reply:
![1670297948219](https://user-images.githubusercontent.com/45089228/205807214-bb27d475-2e0e-44ba-8ba7-adb27b493b30.png)